### DISCUSSION TITLE
Can not find the bootable ssd after installation

### DISCUSSION THREAD

**Discussion by yuchong-li** [osy86/HaC-Mini#936] [Q&A] [answered]
Hi guys, 

I've finished the installation process now and everything runs smoothly.

But I can not find the bootable SSD after installation even the disk utility told me the ssd is bootable.

No bootable options in bois, even the shell command does not work (fs0:\EFI\BOOT\BOOTx64.efi)

So I haven't actually booted into the fresh OSX yet.

And I tried to select boot from Catalina in the USB installer startup disk option but the system will always boot the usb rather than the SSD even the Catalina startup disk has been selected.

![image](https://user-images.githubusercontent.com/49866477/135737166-be6e8c6e-cdec-4148-9bc0-82b8d1656aec.jpeg)
![image](https://user-images.githubusercontent.com/49866477/135737168-ecdf3e98-5e17-4b01-8050-e0e0a332010c.jpeg)

**Comment by osy** [comment, marked as answer]
Did you run the HaC mini installer on your fresh installed OS like it said in the guide?

**Reply by yuchong-li** [reply]
Hi! Yes! I have used the HacMini.pkg to modified the usb before I plugged in.

I selected the patch osx installer and 1820A options.

**Reply by osy** [reply]
Did you read what I said? Run it on your macos installation AFTER it’s installed.

**Reply by yuchong-li** [reply]
Sorry for my misunderstanding due to my poor English. 
I haven't booted into the fresh OSX yet because there is no way to boot it.

**Reply by osy** [reply]
Use the USB to boot into the new installation. Then install hac mini in the new os.

**Reply by yuchong-li** [reply]
Hi, I'm not sure how to boot from USB sorry for the stupid question I asked. 
Is that means I need to keep pressing or holding the Alt key when the intel logo showing to get into the boot picker page? Then pick the OS X? 

I'm still trying to do it. 
Thank you so much for giving me a direction to research. 

//Update

Sorry, I did not be very clear before, after the installation the NUC directly boot into the USB installation page (macOS utilities page with 4 options and the second option is "install macOS" rather than "reinstall macOS") again rather than boot into the macOS, so I am doubting the installation has not finished from the start.

Reference:(the Nuc will boot into this page after installation finished) 
![image](https://user-images.githubusercontent.com/49866477/135752850-6c56441e-e44a-41a6-8fd8-8607d6c933b4.jpeg)

At this stage every time when I switch the NUC on (with USB), it will directly boot into the GUI OSX installer selection page which contains two options "install Catalina" and "reset NVRAM". 

There is no command-line page that I can select the boot macOS/installer etc like all video tutorials that I can found online.

So every time I can only boot into installation or shell. 

In the USB Installation page, as I said before, I can see the Catalina option in the startup disk selection but I can not boot it even it has been selected before I restart the system.

If I pressing F10 when the NUC powering on, there will be 2 options "OpenCore" and "Shell".

I do not have an internet connection during the entire installation process on NUC, maybe some issues are caused by this?

Sorry for the dumpy questions I asked, and really appreciate your help.

**Reply by pkoshevoy** [reply]
I'm having the same problem -- HaC Mini 3.1.1, NUC8i7HVK, BIOS version HN0067 (and I haven't been able to downgrade it)

Install appears to be going well, but after the 2nd reboot there is no option to boot from the internal drive (Samsung 980 SSD in my case).  Worse, afterwards I can't boot from the external USB HDD that has Kubuntu installed, which I was able to boot from before -- the only boot option available from the F10 boot picker is opencore and that doesn't work.  The drive wasn't even connected to the NUC during macOS installation, so I have no idea why it would be affected.  To resolve the external USB HDD Kubuntu boot issue I had to reinstall the original NVMe SSD drive that has Win10, and used bcdedit to remove all the opencore boot manager entries... then reboot without any NVMe SSD installed and after a couple of tries Kubuntu is able to boot again.

That doesn't really help me with the (incomplete?) macOS installation on the 980 SSD.  I've tried copying the EFI folder from the installer USB thumb drive to the 980 SSD EFI partition -- it still doesn't boot, the only option it shows afterwards is Reset NVRAM.

**Reply by pkoshevoy** [reply]
I've managed to install successfully and boot the installed macOS 10.15.7 by using HaC Mini v 2.17, with default settings except for:
- enable Patch OSX installer
- enable Verbose boot
- enable Debug mode
- enable Show boot menu

**Reply by yuchong-li** [reply]
Thank you so much! 
I've been working it out as well based on your configuration!

**Reply by yuchong-li** [reply]
> I'm having the same problem -- HaC Mini 3.1.1, NUC8i7HVK, BIOS version HN0067 (and I haven't been able to downgrade it)
> 
> 
> 
> Install appears to be going well, but after the 2nd reboot there is no option to boot from the internal drive (Samsung 980 SSD in my case).  Worse, afterwards I can't boot from the external USB HDD that has Kubuntu installed, which I was able to boot from before -- the only boot option available from the F10 boot picker is opencore and that doesn't work.  The drive wasn't even connected to the NUC during macOS installation, so I have no idea why it would be affected.  To resolve the external USB HDD Kubuntu boot issue I had to reinstall the original NVMe SSD drive that has Win10, and used bcdedit to remove all the opencore boot manager entries... then reboot without any NVMe SSD installed and after a couple of tries Kubuntu is able to boot again.
> 
> 
> 
> That doesn't really help me with the (incomplete?) macOS installation on the 980 SSD.  I've tried copying the EFI folder from the installer USB thumb drive to the 980 SSD EFI partition -- it still doesn't boot, the only option it shows afterwards is Reset NVRAM.
> 
> 
Hi, 
To update this thread.

I've using an enclosure to check the content on it after I installed 10.15.7 by using 3.1.2 hacmini pkg.

Obviously, there was no EFI partition been created after the installation. (No sure about the reason since the erase configuration was selected correctly)
This should be the reason for all above.

Currently, I'm still finding the way to manually create the EFI partition on the external drive.( no clue yet )

Anyway, the reason is founded.

![image](https://user-images.githubusercontent.com/49866477/136018138-cbdac900-8e16-4de9-8b2f-dbbade83bec8.jpeg)![image](https://user-images.githubusercontent.com/49866477/136018153-b1bf95e4-d9f5-4edb-84e2-6f351d8ed025.jpeg)

**Reply by fucongwanglucid** [reply]
Nice find! I had the same problem upgrading from 3.0.1 to 3.1.2.

**Reply by yuchong-li** [reply]
> I've managed to install successfully and boot the installed macOS 10.15.7 by using HaC Mini v 2.17, with default settings except for:
> 
> - enable Patch OSX installer
> 
> - enable Verbose boot
> 
> - enable Debug mode
> 
> - enable Show boot menu

Hi!!!
I am happy to say that V3.1.2 works with Big Sur installer.

With 

- 	enable Verbose boot

- 	enable Debug mode

- 	enable Show boot menu

If you need a big sur installation DMG, leave me your email address~

Good luck.

**Reply by yuchong-li** [reply]
> Nice find! I had the same problem upgrading from 3.0.1 to 3.1.2.

3.1.2 works with Big Sur USB driver on my side!!!

With 

- enable Verbose boot

- enable Debug mode

- enable Show boot menu

**Reply by x2358473** [reply]
I'm also the 10.15.7 system。3.1.2 the boot system option cannot be displayed after the boot installation of. Is there a solution. I like the 10.15 system very much and want to continue to update the boot on this system and use it. I'm from China. English is not very good. Sorry.

**Reply by yuchong-li** [reply]
> I'm also the 10.15.7 system。3.1.2 the boot system option cannot be displayed after the boot installation of. Is there a solution. I like the 10.15 system very much and want to continue to update the boot on this system and use it. I'm from China. English is not very good. Sorry.

我没找到能够让 3.1.2 正常跑 10.15.7 的办法，我是直接用装了Big Sur installation的U盘跑3.1.2解决。

I did not find a way to let 10.15.7 work with V3.1.2, so I just run the 3.1.2 with a USB that has Big Sur Installation on it to solve it.

**Reply by x2358473** [reply]
> > 我也是10.15.7系统。3.1.2的boot system选项在boot安装后无法显示。有没有解决办法。非常喜欢10.15系统，想继续更新这个系统上的boot，用上。我来自中国。英语不太好。对不起。
> 
> 我没找到能够让 3.1.2 正常跑 10.15.7 的办法，我是直接用装了 Big Sur 安装的 U盘跑 3.1.2 解决。
> 
> 我没有找到让 10.15.7 与 V3.1.2 一起工作的方法，所以我只是运行带有 Big Sur 安装的 USB 的 3.1.2 来解决它。

Thank you!  Then I can only continue to use 3.0.1 boot.
